### ISSUE TITLE
Cache post requests

### ISSUE THREAD

**Issue by hakhap** [akhoury/request-promise-cache#1]
I've been using request-promise-cache for a while now and while running some tests noted that the POST request are not being cached anymore.

I use request-promise-cache to cache API responses, various timeouts depending on the data. Will POST caching be added again?  OR is there another way to get it working?

**Comment by akhoury** [comment]
I don't think we should cache POST requests. POST is "usually" used to "modify/create" an entity on the server, so you'd expect to get the "modified/new" entity back, not the cached one. See https://stackoverflow.com/a/626083/493756 

Granted, we can still cache them, but I don't think it's a good idea. 

Can you tell me your use case for caching POST requests?

**Comment by franciscop** [comment]
Agree with @akhoury, IMHO it only makes sense to cache GET requests.

**Comment by seeARMS** [comment]
Here is a quick hack I put together to cache results of POSTs: https://github.com/seeARMS/request-promise-cache/

My use-case: I needed to POST something to login, but wanted to cache the resulting access token without explicitly storing it somewhere.

**Comment by BrkDalkiran** [comment]
@akhoury `I don't think we should cache POST requests. POST is "usually" used to "modify/create" an entity on the server`

You're right but it doesn't always happen. I'm using too many query parameters and exceeding the 2048 limit. So i use POST method. Is it possible to cache this?
